sdv/timeseries/deepecho.py

```diff
@@ -67,7 +67,8 @@ def _fit(self, timeseries_data):
 
         data_types = list()
         context_types = list()
-        for field, meta in self._metadata.get_fields().items():
+        for field in self._entity_columns + self._data_columns:
```

Comment:
I think this will possibly not work because the order of the columns will be altered, and also we would be missing the `context_columns`.

If the order of the key/value pairs from the `self._metadata.get_fields()` is the problem, maybe a possibility would be to just iterate over `self._output_columns` (which is the list of columns from the input data)?

Then, in order to solve the `sequence_index` problem, we could change line 74 (from the old code):

```python
                if field == self._sequence_index:
                    data_types.append('continuous')
```

to

```python
                if field == self._sequence_index:
                    data_types.extend(['continuous', 'continuous'])
```

And then just remove line 82 (from the old code).